x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseService.java

```diff
@@ -352,17 +352,10 @@ protected ClusterStateUpdateResponse newResponse(boolean acknowledged) {
 
                 @Override
                 public ClusterState execute(ClusterState currentState) throws Exception {
-                    MetaData metaData = currentState.metaData();
-                    final LicensesMetaData currentLicenses = metaData.custom(LicensesMetaData.TYPE);
-                    if (currentLicenses.getLicense() != LicensesMetaData.LICENSE_TOMBSTONE) {
-                        MetaData.Builder mdBuilder = MetaData.builder(currentState.metaData());
-                        LicensesMetaData newMetadata = new LicensesMetaData(LicensesMetaData.LICENSE_TOMBSTONE,
-                            currentLicenses.getMostRecentTrialVersion());
-                        mdBuilder.putCustom(LicensesMetaData.TYPE, newMetadata);
-                        return ClusterState.builder(currentState).metaData(mdBuilder).build();
-                    } else {
-                        return currentState;
-                    }
+                    final StartBasicClusterTask startBasicClusterTask =
+                        new StartBasicClusterTask(logger, clusterService.getClusterName().value(), clock,
+                            new PostStartBasicRequest().acknowledge(true), null);
+                    return startBasicClusterTask.execute(currentState);
```

Comment:
This feels weird to me.
We run a task, that just creates another task and executes that, and it feels like that's just so we can change the response object.

Why not change this to something like:
```
public void removeLicense(final DeleteLicenseRequest request, final ActionListener<PostStartBasicResponse> listener) {
      PostStartBasicRequest startBasicRequest = new PostStartBasicRequest().acknowledge(true);
      startBasicLicense(startBasicRequest, listener); 
}
```
and then have the Transport action translate the `PostStartBasicResponse` into the `AcknowledgedResponse` it needs to keep compatibilty with the API.

Reply:
Makes sense. I'll update it with a small change: instead of delegating to `startBasicLicense()` method, I'll keep the task submission part so that the context is not lost, i.e. `clusterService.submitStateUpdateTask("delete license", new StartBasicClusterTask(...))`